koku/cost_models/serializers.py

```diff
@@ -54,6 +55,44 @@ class MarkupSerializer(serializers.Serializer):
     unit = serializers.ChoiceField(choices=MARKUP_CHOICES, required=False)
 
 
+class DistributionSerializer(BaseSerializer):
+    """Serializer for distribution options"""
+
+    DISTRIBUTION_OPTIONS = ("distribute_type", "worker_cost", "platform_cost")
+
+    distribute_type = serializers.ChoiceField(choices=metric_constants.DISTRIBUTION_CHOICES, required=False)
+    platform_cost = serializers.BooleanField(required=False)
+    worker_cost = serializers.BooleanField(required=False)
+
+    def validate_distribute_type(self, distribute_type):
+        """Run validation for distribute type."""
+        distrib_choice_list = [choice[0] for choice in metric_constants.DISTRIBUTION_CHOICES]
+        if distribute_type not in distrib_choice_list:
+            error_msg = f"{distribute_type} is an invaild distribution type"
+            raise serializers.ValidationError(error_msg)
+        return distribute_type
+
+    def validate(self, data):
+        """Run validation for distribution options."""
+
+        if not any(ele in data.keys() for ele in self.DISTRIBUTION_OPTIONS):
+            data = data.get("distribution_info", {})
+        if data.keys() and not (sorted(self.DISTRIBUTION_OPTIONS) == sorted(data.keys())):
+            distribution_info_str = ", ".join(str(option) for option in self.DISTRIBUTION_OPTIONS)
+            error_msg = f"Missing distribution information: one of {distribution_info_str}"
+            raise serializers.ValidationError(error_msg)
+        else:
+            _distribute_type = data.get("distribute_type", metric_constants.CPU_DISTRIBUTION)
+            valid_distribute_type = self.validate_distribute_type(_distribute_type)
+            if data == {}:
```

Comment:
It's more idiomatic to use a truth value test rather than comparison.
```suggestion
            if not data:
```